src/main/scala/com.socrata.geoexport/encoders/ShapefileEncoder.scala

```diff
@@ -0,0 +1,201 @@
+        new File(shpFile.getParent, s"${name}.${other}")
+      }
+    }
+  }
+
+  private def toIntermediaryReps(schema: Seq[SoQLColumn]) = schema.map(ShapeRep.repFor(_, ShapefileRepMapper))
+
+  private def buildFeatureType(schema: IntermediarySchema): SimpleFeatureType = {
+    val builder = new SimpleFeatureTypeBuilder()
+
+    builder.setCRS(DefaultGeographicCRS.WGS84)
+    builder.setName("FeatureType")
+
+    val addedNames = schema
+    .filter {
+      case r: IDRep => false
+      case _ => true
+    }.flatMap { rep =>
+      val restrictions = seqAsJavaList(List[Filter]())
+
+      val attrNames = rep.toAttrNames
+      val bindings = rep.toAttrBindings
+
+      if(attrNames.size != bindings.size) {
+        throw new Exception("Attribute names:bindings mapping must be 1:1 in the intermediary representation")
+      }
+
+      attrNames.zip(bindings).foreach { case (attrName, binding) =>
+        // geotools is dumb
+        // scalastyle:off
+        if (classOf[Geometry].isAssignableFrom(binding)) {
+          val attrName = new NameImpl("the_geom")
+          val geomType = new GeometryTypeImpl(attrName, binding, DefaultGeographicCRS.WGS84, true, false, seqAsJavaList(List()), null, null)
+          val geomDescriptor = new GeometryDescriptorImpl(geomType, attrName, Int.MinValue, Int.MaxValue, true, null)
+          builder.add(geomDescriptor)
+        } else {
+          val bindingName = new NameImpl(binding.toString())
+          val attrType = new AttributeTypeImpl(bindingName, binding, true, false, restrictions, null, null)
+          val descriptor = new AttributeDescriptorImpl(attrType, new NameImpl(attrName), Int.MinValue, Int.MaxValue, true, null)
+          builder.add(descriptor)
+        }
+        // scalastyle:on
+      }
+      attrNames
+    }
+    log.debug(s"Added names ${addedNames.size}  ${addedNames}")
+    builder.buildFeatureType()
+  }
+
+  // scalastyle:off
+  private def buildFeature(
+    featureId: Int,
+    featureType: SimpleFeatureType,
+    attributes: IntermediaryValues): SimpleFeature = {
+
+    val id = new FeatureIdImpl(featureId.toString)
+
+    val nonIds = attributes.filter { v =>
+      v match {
+        case (_: SoQLID, _) => false
+        case _ => true
+      }
+    }
+    // what in the actual .... there has to be a better way to do this...
+    // except not because dependent types
+    val values = seqAsJavaList(nonIds.flatMap {
+      thing => ShapefileRepMapper.toAttr(thing)
+    })
+    new SimpleFeatureImpl(values, featureType, id)
+  }
+  // scalastyle:on
+  private def addFeatures(layer: SoQLPackIterator, featureType: SimpleFeatureType, file: File,
+                          dataStore: DataStore, reps:Seq[ShapeRep[_ <: SoQLValue]]) = {
+    dataStore.getFeatureSource((dataStore.getTypeNames.toList)(0)) match {
+      case featureStore: SimpleFeatureStore =>
+
+        for { trans <- managed(new DefaultTransaction(file.getName)) } {
```

Comment:
Instead of the single item for loop, you should use SimpleArm's `using` function.